src/ip.py

```diff
@@ -747,6 +781,9 @@ def main(argv):
         elif argv[0].startswith("-color"):
             perror("iproute2mac: Color option is not implemented")
             argv.pop(0)
+        elif "-json".startswith(argv[0]):
```

Comment:
Can you please add the -pretty option as well?  e.g. `ip -pretty -json link show`.

It should be easy to implement using e.g. `json.dumps(result, indent=4)`